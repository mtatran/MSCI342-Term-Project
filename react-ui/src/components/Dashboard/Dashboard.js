import React, { Component, Fragment, Router } from "react";
import { withRouter } from "react-router";
import { Redirect, Switch, Route, Link } from "react-router-dom";

import "./Dashboard.scss";

import logo from '../../logo.svg'
import Profile from "../Layouts/Profile";
import Home from "../Layouts/Home";
import NotFound from "../Layouts/404";
import CreateAppointment from "../Layouts/CreateAppointment";
import Availability from "../Layouts/Availability";

import UserTypes from '../../constants/userTypes.json';

class Dashboard extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isLogout: false,
      email: this.props.match.params.email,
      userType: this.props.match.params.userType,
      firstName: this.props.match.params.name,
      personId: this.props.match.params.personId,
      accessToken: this.props.match.params.accessToken,
    };
  }

  signOut = () => {
    localStorage.removeItem("token");
    this.setState({
      isLogout: true
    });
  };

  render() {
    // Returns the login screen if the user decides to log out of the system.
    if (this.state.isLogout) return <Redirect to="/login" />;

    // Else it will display the appropriate header based on the user type.
    const { email, firstName, personId, userType, accessToken } = this.state;

    return (
      <Fragment>
        <header className={userType}>
          <div className="column left">
            <img src={logo} className="App-logo" alt="logo" />
          </div>

          <div className="column right">
            <ul className="topnav">
              <li>
                <Link to={`/dashboard`}>Home</Link>
              </li>
              <li>
                <Link to={`/dashboard/Availability`}>Availability</Link>
              </li>
              <li>
                <Link to={`/dashboard/Profile/${email}`}>Profile</Link>
              </li>
              {/* Only display the book appointment form if the user is a student. */}
              {userType === UserTypes.student &&
                <li>
                  <Link to={`/dashboard/CreateAppointment/${email}`}>Create Appointment</Link>
                </li>
                
              }
              <li className="push-right">
                <button onClick={this.signOut} href="#">Sign Out</button>
              </li>
            </ul>
          </div>
        </header>

        <div>
          <main role="main">
            <div className="main">
              <Switch>
                <Route path={`/dashboard/Profile`}>
                  <Profile name={firstName} />
                </Route>
                <Route path={`/dashboard/CreateAppointment`}>
                  <CreateAppointment name={personId} />
                </Route>
                <Route path={`/dashboard/Availability`}>
                  <Availability />
                </Route>
                <Route exact path={`${this.props.match.path}`}>
                  <Home userType={userType} personId={personId} accessToken={accessToken} />
                </Route>
                <Route path="*">
                  <NotFound />
                </Route>
              </Switch>
            </div>
          </main>
        </div>
      </Fragment>
    );
  }
}

export default withRouter(Dashboard);
