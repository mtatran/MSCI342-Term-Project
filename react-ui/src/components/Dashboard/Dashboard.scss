@import "../Common/ColorSchema.scss";
  
.worker {
  background-color: $worker-color;
  height: 70px;
  padding: 5px;

  /* Change the color of links on hover */
  li:hover, button {
    background-color: $worker-hover-color;
  }

  /* Add a color to the active/current link */
  li.active, button {
    background-color: $worker-active-color;
  }
}

.student {
  background-color: $student-color;
  height: 70px;
  padding: 5px;

  /* Change the color of links on hover */
  li:hover, button {
  background-color: $student-hover-color;
  }

  /* Add a color to the active/current link */
  li.active, button {
    background-color: $student-active-color;
  }
}

/* Create two unequal columns that floats next to each other */
.column {
  float: left;
}

.left {
  width: 25%;
}

.right {
  width: 75%;
  display: flex;
  justify-content: flex-end;
  align-items: center;
}

/* Style the links inside the navigation bar */
.topnav {
  list-style-type: none;  

  li {
    float: left;
    text-align: center;
    padding: 14px 16px;
    font-size: 17px;
  }

  a:link, a:visited, button {
    color: $navbar-text-color;
    text-decoration: none;
    display: inline-block;
  }

  button {
    border: none;
    background-color: inherit;
    font-size: 17px;
  }
}
