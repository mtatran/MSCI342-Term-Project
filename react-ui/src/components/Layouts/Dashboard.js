import React, { Component, Fragment } from "react";
import { Redirect, Switch, Route, Link } from "react-router-dom";
import { withRouter } from "react-router";
import Page2 from "./Page2";
import IndexDashboard from "./IndexDashboard";
import NotFound from "./404";
import Header from "../Header"
import CreateAppointment from "./CreateAppointment"



class Dashboard extends Component { 
  constructor(props) {
    super(props);
    this.state = {
      islogout: false,
      email: this.props.match.params.email,
      type : this.props.match.params.type,
      firstName : this.props.match.params.name,
    };
  }

  componentDidMount() {
    console.log(this.state.firstName);
  }

  signOut = () => {
    localStorage.removeItem("token");
    this.setState({
      islogout: true
    });
  };

  render() {
    if (this.state.islogout) {
      return <Redirect to="/login" />;
    }
    let header;
    if(this.state.type ==="student"){
      header = <Header color="red"/>
    } else {
      header = <Header color="purple"/>
    }
   
    const { match } = this.props;
    return (
      <Fragment>
        {header}
        <br>
        </br>
        <br>
        </br>
        <br>
        </br>
        <div>
        <ul>
          <li>
            <Link to={`/Dashboard/`}>Home</Link>
          </li>
          <li>
            <Link to={`/Dashboard/page2/${this.state.email}`}>Page 2</Link>
          </li>
          <li>
            <Link to={`/Dashboard/CreateAppointment/${this.state.email}`}>Create Appointment</Link>
          </li>

          <li>
            <Link to ={`/workerinputavailability/${this.state.email}`}>Worker Input Time Page</Link>

          </li>
    
          <li className="push-right">
            <button onClick={this.signOut} href="#">
              Sign Out
            </button>
          </li>
        </ul>
        <main role="main">
          <div className="main">
            <Switch>
              <Route path={`/Dashboard/page2`}>
                <Page2 name={this.state.firstName}/>
              </Route>
      
              <Route path={`/Dashboard/CreateAppointment`}>
                <CreateAppointment name={this.state.personId}/>
              </Route>
              <Route exact path={`${match.path}`}>
                <IndexDashboard/>
              </Route>
              <Route path="*">
                <NotFound />
              </Route>
            </Switch>
          </div>
        </main>
      </div>
      </Fragment>
    );
     
  }
}

export default withRouter(Dashboard);
