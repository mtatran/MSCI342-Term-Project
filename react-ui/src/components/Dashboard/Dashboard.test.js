import React from "react";
import { BrowserRouter } from "react-router-dom";
import { create } from "react-test-renderer";
import Dashboard from './Dashboard';

describe('Dashboard Snapshot', () => {
  test('snapshot renders', () => {
    const component = create(
    <BrowserRouter>
    <Dashboard />
    </BrowserRouter>  );
    expect(component.toJSON()).toMatchSnapshot();
  });
});