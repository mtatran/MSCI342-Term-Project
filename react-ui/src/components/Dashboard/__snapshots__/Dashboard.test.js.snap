// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Snapshot snapshot renders 1`] = `
Array [
  <header>
    <div
      className="column left"
    >
      <img
        alt="logo"
        className="App-logo"
        src="logo.svg"
      />
    </div>
    <div
      className="column right"
    >
      <ul
        className="topnav"
      >
        <li>
          <a
            href="/dashboard"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li
          className="push-right"
        >
          <button
            href="#"
            onClick={[Function]}
          >
            Sign Out
          </button>
        </li>
      </ul>
    </div>
  </header>,
  <div>
    <main
      role="main"
    >
      <div
        className="main"
      >
        <div>
          <p
            className="Title-Style"
          >
            Welcome
          </p>
        </div>
        <h3>
          No Upcoming Appointments!
        </h3>
      </div>
    </main>
  </div>,
]
`;
